python2/koans/about_string_manipulation.py

```diff
@@ -51,26 +51,26 @@ def test_strings_can_be_split_with_different_patterns(self):
 
         words = pattern.split(string)
 
-        self.assertEqual([__, __, __, __], words)
+        self.assertEqual(["the", "rain", "in", "spain"], words)
```

Comment:
> 50 pattern = re.compile(',|;')
> I'm also not sure how this works

This is an example of a Regular Expression  (`regexp`, for short). Notice that the [`re` module](https://docs.python.org/2/library/re.html) is imported first. See this for an [introduction](https://www.tutorialspoint.com/python/python_reg_expressions.htm). You'll see more of this later, don't worry.

In short, `split` splits the string into sub-strings separated by instances of `pattern`, whereas the pattern `,|;` matches whenever you find a `,` OR a `;`.

Reply:
Q: Say I have a string with a human readable list of ingredients: `"flour, tomato souse, 2 eggs, cheese, ham, and olives"`. Find a regex pattern that will split that string into a list of ingredients for a machine, such as `[ "flour", "tomato souse", "2 eggs", "cheese", "ham", "olives"]`

Reply:
Could that be `pattern=re.compile(,)`

Reply:
Nop, there are spaces and and 'and" too! You'll need a pattern that matches those too.

Reply:
I think my best option is `pattern=re.compile(,|and)`, because including the space separates the `2` and the `eggs`